### ISSUE TITLE
bagel结果无法复现

### ISSUE THREAD

**Issue by Hungryyan1** [OneIG-Bench/OneIG-Benchmark#8]
您好，我想复现官方的bagel评测结果（EN版本），但是跑下来发现都没有对上，尤其在text上差异巨大，请问可能是什么原因呢？我是参考bagel官方写的推理脚本，可否提供你们测评bagel时使用的推理脚本呢？

测评结果：
,alignment
Bagel_base,0.8142395730023245

,anime,human,object,text,reasoning,total average
Bagel_base,0.1733846374514963,0.17218033778048597,0.20874406898870435,0.2612027086317539,0.4797389078140259,0.2568520649912813

,reasoning
Bagel_base,0.20377427842881946

,style
Bagel_base,0.3860663788376801

,ED,CR,WAC,text score
Bagel_base,83.355,0.07,0.20714489446662865,0.38537753290787224

<img width="534" height="178" alt="Image" src="https://github.com/user-attachments/assets/14ba2ba5-3667-4a7f-a508-dd4878ffdf5d" />
我检查了bagel的参数与论文中配置一致，评测也使用的是qwen2.5-vl-7b，没有改动

**Comment by Hungryyan1** [comment]
这是我的环境情况，我是按照官方按照的
Package                  Version
------------------------ ------------
accelerate               1.4.0
av                       16.0.1
bcrypt                   5.0.0
boto3                    1.42.3
botocore                 1.42.3
certifi                  2025.11.12
cffi                     2.0.0
charset-normalizer       3.4.4
cryptography             46.0.3
dreamsim                 0.2.1
einops                   0.8.1
filelock                 3.19.1
flash-attn               2.7.3
fsspec                   2025.9.0
ftfy                     6.3.1
hf-xet                   1.2.0
huggingface-hub          0.36.0
idna                     3.11
Jinja2                   3.1.6
jmespath                 1.0.1
MarkupSafe               2.1.5
megfile                  5.0.2
mpmath                   1.3.0
networkx                 3.3
numpy                    2.1.2
nvidia-cublas-cu11       11.11.3.6
nvidia-cuda-cupti-cu11   11.8.87
nvidia-cuda-nvrtc-cu11   11.8.89
nvidia-cuda-runtime-cu11 11.8.89
nvidia-cudnn-cu11        9.1.0.70
nvidia-cufft-cu11        10.9.0.58
nvidia-curand-cu11       10.3.0.86
nvidia-cusolver-cu11     11.4.1.48
nvidia-cusparse-cu11     11.7.5.86
nvidia-nccl-cu11         2.21.5
nvidia-nvtx-cu11         11.8.86
open_clip_torch          3.2.0
openai-clip              1.0.1
packaging                25.0
pandas                   2.3.3
paramiko                 3.5.1
peft                     0.17.1
pillow                   11.3.0
pip                      25.3
psutil                   7.1.3
pycparser                2.23
PyNaCl                   1.6.1
python-dateutil          2.9.0.post0
pytz                     2025.2
PyYAML                   6.0.3
qwen-vl-utils            0.0.14
regex                    2025.11.3
requests                 2.32.5
s3transfer               0.16.0
safetensors              0.7.0
scipy                    1.15.3
setuptools               80.9.0
six                      1.17.0
sympy                    1.13.1
timm                     1.0.22
tokenizers               0.21.4
torch                    2.6.0+cu118
torchvision              0.21.0+cu118
tqdm                     4.67.1
transformers             4.50.0
triton                   3.2.0
typing_extensions        4.15.0
tzdata                   2025.2
urllib3                  2.5.0
wcwidth                  0.2.14
wheel                    0.45.1

**Comment by PoiuySCFW** [comment]
Hello, Hungryyan1. I'm sorry to hear about that. Attached is the inference script we used previously, which you can use as a reference.

```
import os
import time
import random
import numpy as np
import torch
from accelerate import infer_auto_device_map, load_checkpoint_and_dispatch, init_empty_weights
from inferencer import InterleaveInferencer

from data.transforms import ImageTransform
from data.data_utils import add_special_tokens
from modeling.bagel import (
    BagelConfig, Bagel, Qwen2Config, Qwen2ForCausalLM, SiglipVisionConfig, SiglipVisionModel
)
from modeling.qwen2 import Qwen2Tokenizer
from modeling.autoencoder import load_ae

model_path = "ByteDance-Seed/BAGEL-7B-MoT"  # Download from https://huggingface.co/ByteDance-Seed/BAGEL-7B-MoT

# LLM config preparing
llm_config = Qwen2Config.from_json_file(os.path.join(model_path, "llm_config.json"))
llm_config.qk_norm = True
llm_config.tie_word_embeddings = False
llm_config.layer_module = "Qwen2MoTDecoderLayer"

# ViT config preparing
vit_config = SiglipVisionConfig.from_json_file(os.path.join(model_path, "vit_config.json"))
vit_config.rope = False
vit_config.num_hidden_layers = vit_config.num_hidden_layers - 1

# VAE loading
vae_model, vae_config = load_ae(local_path=os.path.join(model_path, "ae.safetensors"))

# Bagel config preparing
config = BagelConfig(
    visual_gen=True,
    visual_und=True,
    llm_config=llm_config, 
    vit_config=vit_config,
    vae_config=vae_config,
    vit_max_num_patch_per_side=70,
    connector_act='gelu_pytorch_tanh',
    latent_patch_size=2,
    max_latent_size=64,
)

with init_empty_weights():
    language_model = Qwen2ForCausalLM(llm_config)
    vit_model      = SiglipVisionModel(vit_config)
    model          = Bagel(language_model, vit_model, config)
    model.vit_model.vision_model.embeddings.convert_conv2d_to_linear(vit_config, meta=True)

# Tokenizer Preparing
tokenizer = Qwen2Tokenizer.from_pretrained(model_path)
tokenizer, new_token_ids, _ = add_special_tokens(tokenizer)

# Image Transform Preparing
vae_transform = ImageTransform(1024, 512, 16)
vit_transform = ImageTransform(980, 224, 14)

max_mem_per_gpu = "80GiB"  # Modify it according to your GPU setting. On an A100, 80 GiB is sufficient to load on a single GPU.

device_map = infer_auto_device_map(
    model,
    max_memory={i: max_mem_per_gpu for i in range(torch.cuda.device_count())},
    no_split_module_classes=["Bagel", "Qwen2MoTDecoderLayer"],
)
print(device_map)

same_device_modules = [
    'language_model.model.embed_tokens',
    'time_embedder',
    'latent_pos_embed',
    'vae2llm',
    'llm2vae',
    'connector',
    'vit_pos_embed'
]

if torch.cuda.device_count() == 1:
    first_device = device_map.get(same_device_modules[0], "cuda:0")
    for k in same_device_modules:
        if k in device_map:
            device_map[k] = first_device
        else:
            device_map[k] = "cuda:0"
else:
    first_device = device_map.get(same_device_modules[0])
    for k in same_device_modules:
        if k in device_map:
            device_map[k] = first_device

# Thanks @onion-liu: https://github.com/ByteDance-Seed/Bagel/pull/8
model = load_checkpoint_and_dispatch(
    model,
    checkpoint=os.path.join(model_path, "ema.safetensors"),
    device_map=device_map,
    offload_buffers=True,
    dtype=torch.bfloat16,
    force_hooks=True,
    offload_folder="/tmp/offload"
)

model = model.eval()
print('Model loaded')

inferencer = InterleaveInferencer(
    model=model, 
    vae_model=vae_model, 
    tokenizer=tokenizer, 
    vae_transform=vae_transform, 
    vit_transform=vit_transform, 
    new_token_ids=new_token_ids
)

seed = 42
random.seed(seed)
np.random.seed(seed)
torch.manual_seed(seed)
if torch.cuda.is_available():
    torch.cuda.manual_seed(seed)
    torch.cuda.manual_seed_all(seed)
torch.backends.cudnn.deterministic = True
torch.backends.cudnn.benchmark = False

def bagel_inference(prompt):
    inference_hyper=dict(
        cfg_text_scale=4.0,
        cfg_img_scale=1.0,
        cfg_interval=[0.4, 1.0],
        timestep_shift=3.0,
        num_timesteps=50,
        cfg_renorm_min=1.0,
        cfg_renorm_type="global",
    )
    prompt = prompt
    print(prompt)
    print('-' * 10)
    start_time = time.time()
    output_dict = inferencer(text=prompt, **inference_hyper)
    end_time = time.time()
    print("Inference Time Per Image:", end_time - start_time)
    return output_dict["image"]
```

**Comment by CostaliyA** [comment]
我遇到了类似的问题，这是否是你们的推理脚本相关？

<img width="573" height="567" alt="Image" src="https://github.com/user-attachments/assets/e97ededc-7d09-4347-9426-ac54f6dcde5c" />使用你们提供的脚本运行的text-12 无法正常显示，使用bagel改写的推理脚本的相同图片：

<img width="279" height="253" alt="Image" src="https://github.com/user-attachments/assets/ed7ab80e-fc97-4daa-ac3d-2244d79c7ccb" />
这种无法正常生成图片不是个例

text-8

<img width="567" height="570" alt="Image" src="https://github.com/user-attachments/assets/2a37a93b-eafb-417b-a567-93d309407287" />text-1。感谢您的解答！

**Comment by PoiuySCFW** [comment]
你好，CostaliyA。

很抱歉给你的测试带来了困扰。我们这里的推理脚本是比较早期bagel刚release的推理脚本，与你现在所使用的推理脚本可能存在差异。这应该是结果无法对齐的最大可能原因。

**Comment by PoiuySCFW** [comment]
你好， @Hungryyan1 和 @CostaliyA ，请问两位这边的测试过程中还有别的问题嘛，如果没有的话，我将关闭这个issue，谢谢。
